extensions/bouncer-networks.md

```diff
@@ -0,0 +1,285 @@
+invalid attribute, the server MUST reply with:
+
+    FAIL BOUNCER INVALID_ATTRIBUTE <subcommand> <netid> <attribute> :Invalid attribute value
+
+If the `subcommand` is `ADDNETWORK`, `netid` MUST be set to the special `*`
+value.
+
+#### `READ_ONLY_ATTRIBUTE` error
+
+If a client attempts to change a read-only network attribute using the
+`ADDNETWORK` or `CHANGENETWORK` subcommand, the server MUST reply with:
+
+    FAIL BOUNCER READ_ONLY_ATTRIBUTE <subcommand> <netid> <attribute> :Read-only attribute
+
+If the `subcommand` is `ADDNETWORK`, `netid` MUST be set to the special `*`
+value.
+
+#### `UNKNOWN_ATTRIBUTE` error
+
+If a client sends an `ADDNETWORK` or a `CHANGENETWORK` subcommand with an
+unknown attribute, the server MUST reply with:
+
+    FAIL BOUNCER UNKNOWN_ATTRIBUTE <subcommand> <netid> <attribute> :Unknown attribute
+
+If the `subcommand` is `ADDNETWORK`, `netid` MUST be set to the special `*`
+value.
+
+#### `NEED_ATTRIBUTE` error
+
+If a client sends an `ADDNETWORK` subcommand without a mandatory attribute, the
+server MUST reply with:
+
+    FAIL BOUNCER NEED_ATTRIBUTE ADDNETWORK <attribute> :Missing required attribute
+
+### Standard network attributes
+
+Bouncers MUST recognise the following standard network attributes:
+
+* `name`: the human-readable name for the network.
+* `state` (read-only): one of `connected`, `connecting` or `disconnected`.
+  Indicates the current state of the connection to the upstream network.
+* `host`: the hostname or literal IP address to connect to.
+* `port`: the TCP port to connect to.
+* `tls`: `1` to use a TLS connection, `0` to use a cleartext connection.
+* `nickname`: the nickname to use during registration.
+* `username`: the username to use during registration.
+* `realname`: the realname to use during registration.
+* `pass`: the server password (PASS) to use during registration.
+
+Some networks MAY have some attributes undefined, thus `LISTNETWORKS` replies
+and network notifications MAY only use a subset of these standard attributes.
+Clients MUST handle networks with missing attributes.
+
+### Examples
+
+Binding to a network:
+
+    C: CAP LS 302
+    C: NICK emersion
+    C: USER emersion 0 0 :Simon
+    S: CAP * LS :sasl=PLAIN draft/bouncer-networks draft/bouncer-networks-notify
+    C: CAP REQ :sasl draft/bouncer-networks
+    [SASL authentication]
+    C: BOUNCER BIND b33f
+    C: CAP END
+
+Listing networks:
+
+    C: BOUNCER LISTNETWORKS
+    S: BATCH +asdf draft/bouncer-networks
+    S: @batch=asdf BOUNCER NETWORK b33f name=Libera.Chat;state=connected;host=irc.libera.chat
+    S: @batch=asdf BOUNCER NETWORK f7edfs name=My\sAwesome\sNetwork;state=disconnected
+    S: BATCH -asdf
+
+Adding a new network:
+
+    C: BOUNCER ADDNETWORK name=OFTC;host=irc.oftc.net
+    S: BOUNCER NETWORK 7dwsw name=OFTC;host=irc.oftc.net;state=connecting
+    S: BOUNCER ADDNETWORK 7dwsw
+    S: BOUNCER NETWORK 7dwsw status=connected
```

Comment:
That's not specified anywhere, I surmise this is:
```suggestion
    S: BOUNCER NETWORK 7dwsw state=connected
```

Reply:
Good catch, fixed!